apps/prairielearn/src/migrations/20240221205019_assessment_access_policies.sql

```diff
@@ -0,0 +1,16 @@
+CREATE TABLE IF NOT EXISTS assessment_access_policies (
+  id BIGSERIAL PRIMARY KEY,
+  assessment_id BIGINT NOT NULL REFERENCES assessments (id) ON UPDATE CASCADE ON DELETE CASCADE,
+  user_id BIGINT REFERENCES users (user_id) ON UPDATE CASCADE ON DELETE CASCADE,
+  group_id BIGINT REFERENCES groups (id) ON UPDATE CASCADE ON DELETE CASCADE,
+  start_date TIMESTAMP WITH TIME ZONE NOT NULL,
+  end_date TIMESTAMP WITH TIME ZONE NOT NULL,
+  credit INTEGER NOT NULL,
+  note TEXT,
+  created_by BIGINT NOT NULL REFERENCES users (user_id),
```

Comment:
@mwest1066 I think `ON DELETE SET NULL` is appropriate here, can you confirm?

```suggestion
  created_by BIGINT NOT NULL REFERENCES users (user_id) ON UPDATE CASCADE ON DELETE SET NULL,
```

Reply:
Yep, agreed, we always want to set explicit policies for ON UPDATE and ON DELETE and for `created_by` I think we will always want `SET NULL` to retain the row even if the creator is deleted.

Reply:
Thanks for confirming! @elisacarrillo, can you make this change?

Reply:
Looks like this was changed in 941fae1.

Reply:
Sorry, unsure why my comment is pending